pkg/apis/eventing/v1alpha1/subscription_types.go

```diff
@@ -175,8 +178,25 @@ type SubscriptionStatus struct {
 	// +patchMergeKey=type
 	// +patchStrategy=merge
 	Conditions duckv1alpha1.Conditions `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+	// Subscription might be Subscribable. This depends if there's a Result channel
+	// In that case, this points to that resource.
+	Subscribable duckv1alpha1.Subscribable `json:"subscribable,omitempty"`
 }
 
+const (
+	// SubscriptionConditionReady has status True when all subconditions below have been set to True.
+	SubscriptionConditionReady = duckv1alpha1.ConditionReady
+
+	// SubscriptionReferencesResolved has status True when all the specified references have been successfully
+	// resolved.
+	SubscriptionConditionReferencesResolved duckv1alpha1.ConditionType = "Resolved"
+
+	// SubscriptionConditionFromReady has status True when controller has successfully added a subscription to From
+	// resource.
+	SubscriptionConditionFromReady duckv1alpha1.ConditionType = "FromReady"
```

Comment:
After you add these, it is helpful to expose an `IsReady` method using the Conditions helper to be the judge. 

https://github.com/knative/eventing/pull/457/files#diff-67a9c730f0d592911dfa317323fea9e5R131

Also a `InitializeConditions`

https://github.com/knative/eventing/pull/457/files#diff-67a9c730f0d592911dfa317323fea9e5R137

And then `Mark*` style methods as you need them.